### PR TITLE
T dh 599 Start of Flows and Triggers

### PR DESCRIPTION
**Pull request by byarsawich**
Breaking up this pull request since there is a ton of code in it.

### CODE REVIEW

**Comment by Fitchburgh** [comment]
Also just noticed you misspelled ServiceAppointmentService class.

File: src/classes/AssignedResourceService.cls [hunk 1]
```diff
@@ -3,4 +3,153 @@
...
+				[SELECT Id, ParentRecordId
+				FROM ServiceAppointment
+				WHERE Id IN :saIds AND ParentRecordType = 'WorkOrder']);
+
+		List<String> parentIds = base_toc.ArrayUtils.getField(allSAs.values(), 'ParentRecordId');
+		List<TimeSheetEntry> tseList =
+				[SELECT Id, StartTime, Status, DurationInMinutes, EndTime, TimeSheetId
+				FROM TimeSheetEntry
+				WHERE WorkOrderId IN :parentIds AND TimeSheet.ServiceResourceId IN :srIds
+		AND TimeSheet.Status = 'Current'];
```
[the remark this answers is not in the document]
Comment (byarsawich) [reply src/classes/AssignedResourceService.cls]: added it to my dev box, we'll have to manually add it to pass the test until we have the metadatutils class

File: src/classes/Test_ChatterUtils.cls [hunk 2]
```diff
@@ -0,0 +1,7 @@
+/*
+ * Created by byarsawi on 9/6/2018.
+ */
+@IsTest
+private class Test_ChatterUtils {
+
```
[the remark this answers is not in the document]
Comment (byarsawich) [reply src/classes/Test_ChatterUtils.cls]: Tech debt for now

**Comment by Fitchburgh** [comment]
Should I merge this or wait for 599b?

**Pull request merged**